### PR TITLE
81-product-validation-messages :: added validation to Create and Edit

### LINKED ISSUE #81: Validation Messages on Product Edit
### Feature Description

When a user edits one of their products and the model is invalid, user is currently redirected to the edit page with the original product information and no validation messages.

To resolve, we need to do something like this:
- Update the Create View with a non-editable input field to contain the ProductId
- On the Create Put method, instead of grabbing the Id from the route, grab it from the submitted form
- Update the Create Put method's signature to accept a Product product only
- If model state is invalid, return the Create View and pass in the product object
### Requirement to Fulfill

> Refer to the [project requirements](https://github.com/TeamCharles/bangazon-inc/blob/master/projects/INITIAL_BANGAZON_SITE.md#requirements) before filling out this section.

### PR DESCRIPTION
**Pull request by MattKraatz**
## Status

**READY**
## Documentation

[X] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.
## Description

Overload the Create and Edit methods on the ProductsController so that the view can be re-displayed if the model is invalid.
## Resolves Issue Number

Resolves #81 
## Deploy Notes

``` sh
git pull --prune
git checkout 81-product-validation-messages
dotnet run
```
## Steps to Test or Reproduce
1. Log in
2. Click `Sell a Product`
3. Mess up the form, you should be routed back to the form with validation messages
4. Edit an existing product
5. Mess up the form, you should be routed back to the form with validation messages
6. Make sure you can actually complete adding a new product and committing edits to an existing product
## Impacted Files in Application

List general components of the application that this PR will affect:
- Views/Products/Create
- Views/Products/Edit
- Controllers/ProductsController

### CODE REVIEW

File: Views/Shared/_NavbarView.cshtml [hunk 1]
```diff
@@ -14,7 +14,7 @@
         <div class="navbar-collapse collapse">
             <ul class="nav navbar-nav navbar-right">
                 <li><a asp-area="" asp-controller="User" asp-action="New">Create an Account</a></li>
-                <li><a asp-area="" asp-controller="Products" asp-action="New">Add Product</a></li>
+                <li><a asp-area="" asp-controller="Products" asp-action="New">Sell a Product</a></li>
```
Comment (Anulfo) [inline Views/Shared/_NavbarView.cshtml:17]: The route is pointing to new, but the form is in /products/create. The validation in the form works well!!. LGTM

**Review by Anulfo — COMMENTED** [review]
LGTM

File: Views/Shared/_NavbarView.cshtml (hunk 1 above)
Comment (daynewright) [reply Views/Shared/_NavbarView.cshtml:17]: @MattKraatz I second @Anulfo.  The `asp-action` needs to be changed to `Create`.  Otherwise LGTM.

**Review by daynewright — CHANGES_REQUESTED** [review]
Change `asp-action` to `Create` in `_NavbarView.cshtml`

File: Views/Shared/_NavbarView.cshtml (hunk 1 above)
Comment (MattKraatz) [reply Views/Shared/_NavbarView.cshtml:17]: Great find, thanks!

**Review by daynewright — APPROVED** [review]

**Comment by garrett-vangilder** [comment]
LGTM

**Pull request merged**